assets/objects/object_link_boy/object_link_boy.h

```diff
@@ -0,0 +1,89 @@
+#ifndef OBJECT_LINK_BOY_H
+#define OBJECT_LINK_BOY_H
+
+#include "ultra64.h"
+#include "z64animation.h"
+
+extern u64 gLinkAdultEyesOpenTex[];
+extern u64 gLinkAdultEyesHalfTex[];
+extern u64 gLinkAdultEyesClosedfTex[];
+extern u64 gLinkAdultEyesLeftTex[];
+extern u64 gLinkAdultEyesRightTex[];
+extern u64 gLinkAdultEyesWideTex[];
+extern u64 gLinkAdultEyesDownTex[];
+extern u64 gLinkAdultEyesWincingTex[];
```

Comment:
I missed this in the new asset system review itself, but before we have a ton of committed headers it might be nice to give these arrays sizes so users can take sizeof() of them

Reply:
Only textures?

Personally I'd rather only put sizes when required/useful

Otherwise how I'd like to declare the size would look like:
```c
#include "gbi.h"
#define gLinkAdultEyesOpenTex_WIDTH 64
#define gLinkAdultEyesOpenTex_HEIGHT 32
extern u64 gLinkAdultEyesOpenTex[gLinkAdultEyesOpenTex_WIDTH * gLinkAdultEyesOpenTex_HEIGHT * G_IM_SIZ_16b_BYTES / sizeof(u64)];
```
which is a bit much imo, unless I manage to cope with `extern u64 gLinkAdultEyesOpenTex[0xHEX];` :copium:

Reply:
Width and height macros sound nice to have. Sounds nice for dlists too

Reply:
that definition is a bit much to me, but maybe it can be macroed if we really dont want to just have raw bytes

Reply:
imo we should have sizes for everything even if it's just a raw number. The specific form of how it's written doesn't matter hugely to me since the main motivation is to allow taking `sizeof()`, which is always potentially useful.

Reply:
Ill get over the very long calculation. I think there are many cases where having width and height macros in the header will be useable in code, so they should definitely exist, on top of the sizeof() point

Reply:
I suppose the question Im getting at here is: Do we expose the WIDTH and HEIGHT of all textures (like the generated source does), or move the ones that arent used in code to the C file (like dragorns pr does currently)

if we want to figure this out separately maybe we go with a hex size for the short term

Reply:
Why is sizeof() useful?

Should we declare all resources in the .h then, and not just externally referenced ones?

Reply:
Can't these just use `abcTex_WIDTH * abcTex_HEIGHT` instead of a sizeof?

> any advanced manipulations of textures in code will want it
I understand but then I would only expose what the code needs still

If we expose all sizes and width/height of resources declared in the .h just because it may come in handy, then I don't see why not declare everything in the .h as it may "come in handy" too?

(I'm brainstorming here, not sure what the best course of actions is, just trying to understand your and tharo's points)

Reply:
It seems much more advantageous to just support `sizeof` than not, otherwise if a user has need for sizeof they'll either need to come complain to get it added or peek into the extraction tools themselves. I'd seen complaints of fast64 not exporting sized arrays, so I figure why fall into the same issue in extracted assets.

As for exposing other things that "might be useful", that's out of scope to this consideration.